packages/flutter_tools/lib/src/reporting/events.dart

```diff
@@ -279,14 +277,14 @@ class NullSafetyAnalysisEvent implements UsageEvent {
       return;
     }
     int migrated = 0;
-    LanguageVersion languageVersion;
+    LanguageVersion? languageVersion;
     for (final Package package in packageConfig.packages) {
       if (package.name == currentPackage) {
         languageVersion = package.languageVersion;
       }
       if (package.languageVersion != null &&
```

Comment:
for cases like this consider something like:
```
var packageLanguageVersion = package.languageVersion;
if (packageLanguageVersion  != null && ...)
```

That way you get some type promotion helping you out